ratis-server/src/main/java/org/apache/ratis/server/impl/ServerState.java

```diff
@@ -431,7 +431,7 @@ private SnapshotInfo getLatestSnapshot() {
 
   long getLatestInstalledSnapshotIndex() {
     final TermIndex ti = latestInstalledSnapshot.get();
-    return ti != null? ti.getIndex(): 0L;
+    return ti != null? ti.getIndex(): -1L;
```

Comment:
Let's use `RaftLog.INVALID_LOG_INDEX` instead of -1.
```suggestion
    return ti != null? ti.getIndex(): RaftLog.INVALID_LOG_INDEX;
```